plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -68,70 +98,84 @@ func (r *RBACAuthorizer) Authorize(attr authorizer.Attributes) error {
+		time.Sleep(t)
+	}
 }
 
-func (p policyLoadError) Error() string {
-	if p.line >= 0 {
-		return fmt.Sprintf("error reading policy file %s, line %d: %s: %v", p.path, p.line, string(p.data), p.err)
+func expBackoff(t time.Duration, nRetries int) func() (time.Duration, bool) {
+	return func() (time.Duration, bool) {
+		if nRetries == 0 {
+			return 0, false
+		}
+		nRetries--
+		backoff := t
+		t = t * 2
+		return backoff, true
 	}
-	return fmt.Sprintf("error reading policy file %s: %v", p.path, p.err)
 }
 
-func New(roleRegistry role.Registry, roleBindingRegistry rolebinding.Registry, clusterRoleRegistry clusterrole.Registry, clusterRoleBindingRegistry clusterrolebinding.Registry, path string) (*RBACAuthorizer, error) {
-	// File format is one map per line.  This allows easy concatentation of files,
-	// comments in files, and identification of errors by line number.
-	file, err := os.Open(path)
-	if err != nil {
+func New(roleRegistry role.Registry, roleBindingRegistry rolebinding.Registry, clusterRoleRegistry clusterrole.Registry, clusterRoleBindingRegistry clusterrolebinding.Registry) (*RBACAuthorizer, error) {
+	// Configure context for the global namespace.
+	ctx := api.WithNamespace(api.NewContext(), "")
+
+	var clusterRoles *rbac.ClusterRoleList
+	if err := retry(expBackoff(2*time.Second, 2), func() error {
+		var err error
+		if clusterRoles, err = clusterRoleRegistry.ListClusterRoles(ctx, &api.ListOptions{}); err != nil {
+			return fmt.Errorf("failed to get cluster roles: %v", err)
+		}
+		return nil
+	}); err != nil {
 		return nil, err
 	}
-	defer file.Close()
-
-	scanner := bufio.NewScanner(file)
-	decoder := api.Codecs.UniversalDecoder(groupVersions...)
-	i := 0
-	for scanner.Scan() {
-		i++
-		b := scanner.Bytes()
-
-		// skip comment lines and blank lines
-		trimmed := strings.TrimSpace(string(b))
-		if len(trimmed) == 0 || strings.HasPrefix(trimmed, "#") {
-			continue
-		}
 
-		decodedObj, _, err := decoder.Decode(b, nil, nil)
-		if err != nil {
-			return nil, policyLoadError{path, i, b, err}
+	if len(clusterRoles.Items) == 0 {
+		var clusterRoleBindings *rbac.ClusterRoleBindingList
+		if err := retry(expBackoff(2*time.Second, 2), func() error {
+			var err error
+			if clusterRoleBindings, err = clusterRoleBindingRegistry.ListClusterRoleBindings(ctx, &api.ListOptions{}); err != nil {
+				return fmt.Errorf("failed to get cluster role bindings: %v", err)
+			}
+			return nil
+		}); err != nil {
+			return nil, err
 		}
 
-		switch obj := decodedObj.(type) {
-		case *rbac.Role:
-			if err := roleRegistry.UpdateRole(api.NewContext(), obj); err != nil {
-				return nil, err
-			}
-		case *rbac.RoleBinding:
-			if err := roleBindingRegistry.UpdateRoleBinding(api.NewContext(), obj); err != nil {
-				return nil, err
-			}
-		case *rbac.ClusterRole:
-			if err := clusterRoleRegistry.UpdateClusterRole(api.NewContext(), obj); err != nil {
+		// If there are no items, create initial bootstrapping policies.
```

Comment:
I was thinking in the Authorizer, it would check how may rules there are in total for the whole cluster, and if that is zero, then authorize everything.